cxx/gendb.cc

```diff
@@ -53,7 +54,8 @@ void GenDB::incorporate(
         schema.query.fields.at(query_rel).class_path;
     T_items items =
         sample_entities_relation(prng, schema.query.record_class,
-                                 class_path.cbegin(), class_path.cend(), id);
+                                 class_path.cbegin(), class_path.cend(), id,
```

Comment:
A lot of the machinery in sample_class_ancestors etc is for the purpose of walking the DAG of references to make sure the reference values are coherent (i.e. if Physician 5 went to School 3, that's correct in all samples). Am I correct that when sample_new is false, we never see the same Physician e.g. more than once, so we never actually have to walk the DAG? If so, then I think we should implement this by simply iterating over a relation's domains and pulling sequential values from the domain CRPs rather than adding complexity to sample_class_ancestors and the other methods that are primarily for inference.

Reply:
Added TODO.  I don't think the sample_new = false case is as simple as you make out, if only because it's what pclean_lib::translate_observations did before and that was far from trivial (and it relied on the annotated_domains_for_relations datastructure, which is being eliminated).

Reply:
Looking again at translate_observations, it looks like it used annotated_domains_for_relations to get unique string names for all of the entities, which we don't need to do here since we're working with entity IDs directly, so I think it is pretty simple (and I'd prefer to do it in this PR though a TODO is ok too).